lib/util/signal.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2024 Huawei Cloud Computing Technologies Co., Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import "sync"
+
+type Signal struct {
```

Comment:
some of my thoughts for this type:

Types, functions and etc. in the `util` package should be well designed and reusable, but I don't think `Signal` meet the requirement because `Close` and `ReOpen` are not symmetrical, it is very possible for other developers to make a misuse. And, the name, 'Signal' seems not a good one either, as it hasn't describe the usage, behavior well.

Because of above, personally, I propose move the code to where the type is used.

Reply:
@localvar 
感谢你的宝贵意见。
这个PR可能不是解决问题的好办法。
本质上来说，这个问题的根因是缺乏对后台任务的有效管理，
在 compaction，降采样，分级存储之间互斥，以及他们和删除表，删除DB，shard 过期等行为之间的互斥逻辑变得很复杂且混乱
如果你有兴趣，可以一起探讨研究一个更好的方案

Reply:
I agree with your idea @localvar . However, we do not have a technical solution to solve the root cause of the problem. However, I think we can open a separate issue for this problem. We can discuss it in detail there. I'm going to merge this PR for an upcoming release.